vlog4j-examples/src/main/java/org/semanticweb/vlog4j/examples/core/AddDataFromRdfFile.java

```diff
@@ -0,0 +1,170 @@
+package org.semanticweb.vlog4j.examples.core;
+
+/*-
+ * #%L
+ * VLog4j Examples
+ * %%
+ * Copyright (C) 2018 VLog4j Developers
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import java.io.File;
+import java.io.IOException;
+
+import org.semanticweb.vlog4j.core.model.api.Atom;
+import org.semanticweb.vlog4j.core.model.api.Constant;
+import org.semanticweb.vlog4j.core.model.api.Predicate;
+import org.semanticweb.vlog4j.core.model.api.Rule;
+import org.semanticweb.vlog4j.core.model.api.Variable;
+import org.semanticweb.vlog4j.core.model.implementation.Expressions;
+import org.semanticweb.vlog4j.core.reasoner.Algorithm;
+import org.semanticweb.vlog4j.core.reasoner.DataSource;
+import org.semanticweb.vlog4j.core.reasoner.Reasoner;
+import org.semanticweb.vlog4j.core.reasoner.exceptions.EdbIdbSeparationException;
+import org.semanticweb.vlog4j.core.reasoner.exceptions.IncompatiblePredicateArityException;
+import org.semanticweb.vlog4j.core.reasoner.exceptions.ReasonerStateException;
+import org.semanticweb.vlog4j.core.reasoner.implementation.RdfFileDataSource;
+import org.semanticweb.vlog4j.examples.ExamplesUtils;
+
+/**
+ * This example is an adaptation of {@link AddDataFromCsvFile}. It shows how
+ * facts can be imported from {@code .nt.gz} files, and also how query answers
+ * can be exported to {@code .csv} files. <i>Note that you can also import from
+ * {@code .nt} files.</i>
+ *
+ * For importing, an {@link RdfFileDataSource} that contains a path to the
+ * corresponding {@code .nt.gz} file must be created. An {@code .nt} file
+ * contains facts over one or more predicates in the N-Triples format, while an
+ * {@code .nt.gz} file is the gzipped version of such an {@code .nt} file. For
+ * exporting, a path to the output {@code .csv} file must be specified.
+ *
+ * @author Christian Lewe
+ *
+ */
+public class AddDataFromRdfFile {
+
+	public static void main(final String[] args)
+			throws EdbIdbSeparationException, IOException, ReasonerStateException, IncompatiblePredicateArityException {
+
+		/* 1. Instantiating entities and rules. */
+		final Predicate triplesEDB = Expressions.makePredicate("triplesEDB", 3);
+		final Predicate triplesIDB = Expressions.makePredicate("triplesIDB", 3);
+
```

Comment:
The encoding of the Bicycle example into a program with triples is correct. However, the standard way of handling unary predicates in RDF (ex: bycicle, wheel) is with special predicate "rdf:type" (the long URI being <http://www.w3.org/1999/02/22-rdf-syntax-ns#type>).
For example, to specify that constant <c> is a bicycle, you have triple
<http://an.example/c > < http://www.w3.org/1999/02/22-rdf-syntax-ns#type > < http://an.example/bicycle > .

This way, you don't need the emptyObject, and the atoms in facts and rules become more intuitive, as they respect standard subject ("c")-predicate("has type")-object("bycicle") format. 

Note that VLog reasoner does not have a special interpretation for < http://www.w3.org/1999/02/22-rdf-syntax-ns#type >, and it is treated as any other predicate during reasoning.